### PR TITLE
BF: Use newer call to frombytes, replacing fromstring.

### PR DESCRIPTION
**Pull request by arokem**
This addresses #443 on my machine, but is clearly not the complete solution to this issue.

### CODE REVIEW

**Comment by effigies** [comment]
#443 seems to be introduced in [Pillow v3.0.0+](https://github.com/python-pillow/Pillow/commit/71c95c8e5f3bba1845444a246d04646825e6bab3). As far as I can tell, Pillow is the generally accepted fork of PIL, so following their deprecations seems pretty safe.

That said, it should still be [`data.tostring`](https://docs.scipy.org/doc/numpy/reference/generated/numpy.ndarray.tostring.html), not [`data.tobytes`](https://docs.scipy.org/doc/numpy/reference/generated/numpy.ndarray.tobytes.html). `tostring` was renamed to `tobytes` in numpy 1.9.0+, but `tostring` was left as a compatibility alias.

File: nibabel/dft.py [hunk 1]
```diff
@@ -138,8 +138,8 @@ def as_png(self, index=None, scale_to_slice=True):
             max = data.max()
             data = data * 255 / (max - min)
         data = data.astype(numpy.uint8)
-        im = PIL.Image.fromstring('L', (self.rows, self.columns),
-                                  data.tostring())
+        im = PIL.Image.frombytes('L', (self.rows, self.columns),
+                                  data.tobytes())
```
Comment (effigies) [inline nibabel/dft.py]: PEP8: `data` should align with `'L'`.

**Comment by effigies** [comment]
Not sure the best way to test this. Can't get Travis to install Pillow==3: https://travis-ci.org/effigies/nibabel/jobs/122307466

**Comment by matthew-brett** [comment]
Have a look at http://travis-wheels.scikit-image.org/ for the wheels stored in the 'wheelhouse' we are using.  I just built Pillow 3.2 (using https://github.com/matthew-brett/travis-wheel-builder) - so I think `Pillow==3.2` will work.

Probably best way is to add `pillow` to DEPENDS for the full tests, and then add the earlier pillow version (say 2.6) to the matrix entry with the minimum pydicom version.

**Comment by arokem** [comment]
How's this?

**Comment by effigies** [comment]
LGTM

**Comment by matthew-brett** [comment]
Ariel - can you rebase on master to get rid of unrelated test failures?

Maybe you'll have to use `fromstring` on Pillow < 3?

**Comment by effigies** [comment]
> Maybe you'll have to use fromstring on Pillow < 3?

`fromstring` was [deprecated](https://github.com/python-pillow/Pillow/blob/2.0.0/PIL/Image.py#L1828-L1835) in 2.0.

**Comment by matthew-brett** [comment]
Hum - but I suppose there will be some people still using PIL.

**Comment by matthew-brett** [comment]
Maybe test against PIL on Python 2.7 instead of older Pillow.

**Comment by effigies** [comment]
Travis' default install [supports](https://travis-ci.org/effigies/nibabel/builds/122322097) `PIL.Image.frombytes`, which did not exist in [PIL 1.1.7](https://bitbucket.org/effbot/pil-117/src/f356a1f64271e9d3206fcf9059492ba1c9e163d6/PIL/Image.py), but I can [break the build](https://travis-ci.org/effigies/nibabel/builds/122324272#L1754) by directly [installing](https://github.com/effigies/nibabel/commit/53e05306110e6a90465823ab09be18e6485a0528) PIL.

So, you could do something like:

```Python
import PIL.Image
if hasattr(PIL.Image, 'frombytes'):
    frombytes = PIL.Image.frombytes
else:
    frombytes = PIL.Image.fromstring
...
im = frombytes('L', (self.rows, self.columns), data.tostring())
```

Still... PIL itself hasn't been updated since Py2.6, and has zero Py3 support. Looking around, Linux and Python distributions prefer Pillow to PIL. Supporting ancient PIL only seems to make sense to me if we're going to continue supporting 2.6.

**Comment by matthew-brett** [comment]
OK - I see the PIL doesn't have anything that can be got from pip, so `pip install pil` will always fail.

So, I agree, I think we can forget about PIL support now.

**Comment by effigies** [comment]
Also need to revert 6b9ce76 (unless we want to support PIL without testing it?). Or just reset to 310b6d8 and force push.

Otherwise, still LGTM.

**Comment by arokem** [comment]
Sorry @effigies -- I might be misunderstanding your comment. My understanding is that `Import PIL` will import pillow on systems that don't have PIL installed, so technically we are not supporting PIL here. The commit https://github.com/nipy/nibabel/commit/6b9ce763d25ed1c50cf33f2873132fb6dc535f2e did not introduce a PIL/pillow dependency -- that was there before (for a while, apparently at least since 2011, according to the blame view of the `import PIL` line).

**Comment by effigies** [comment]
Sorry, for the imprecision. Unfortunately Pillow and PIL are both imported as PIL, so it may be unavoidable. When I write PIL, I'm referring to the library that ended at version 1.1.7, not the import.

The question is not whether to have the dependency, but whether to permit a dependency to support the defunct PIL proper or just stick with the actively maintained Pillow. `.fromstring` is a holdover from PIL that Pillow deprecated for `.frombytes` as of 2.0.0 (released March 2013).

If we don't mean to support PIL, which is ancient, then adding a codepath to use `.fromstring` as a backup seems extraneous. BUT, if we are going to have that codepath, it seems to me that we should go ahead and have a PIL-based Travis test. So my preference would be to merge either 6b9ce76 (support PIL 1.1.7 and Pillow >= 1.0) or 310b6d8 (only support Pillow >= 2.0).

That said, I think the code is in a fine state, so I'll leave remaining decisions to @matthew-brett.

**Comment by arokem** [comment]
OK - thanks for clarifying! I will wait for @matthew-brett to weigh in.

**Comment by matthew-brett** [comment]
Looks good to me - thanks.

**Pull request merged**